test/basics.jl

```diff
@@ -0,0 +1,61 @@
+test_string_length = 100
+
+#string_types = [ASCIIStr, LatinStr, UTF8Str, UCS2Str, UTF16Str, UTF32Str, UniStr]
+#string_types = [ASCIIStr, LatinStr, UTF8Str, UCS2Str, UTF16Str, UTF32Str]
+string_types = [ASCIIStr, LatinStr, UTF8Str]
+
+##  create type specific test strings
+test_strings_base = Dict{String, Any}()
+for T in string_types
```

Comment:
Note, these should really be be based on the character types, not the string types.
`ASCIIChr, LatinChr, UCS2Chr, UTF32Chr`
I believe I added a const for those?

Reply:
Change the string construction to use a comprehension, and the loop to be based on the 4 character types.
For this purpose, you actually don't need to make them different types, because you'll be converting them to the testing types anyway, so they can all be made as `String`.